File: src/main/kotlin/de/byteleaf/companyon/user/entity/UserEntity.kt
package de.byteleaf.companyon.user.entity

import de.byteleaf.companyon.common.annotation.NoArgConstructor
import de.byteleaf.companyon.common.entity.BaseEntity
import de.byteleaf.companyon.fileupload.dto.FileMeta
import org.springframework.data.mongodb.core.mapping.Document

@NoArgConstructor
@Document(collection = "users")
data class UserEntity(
        var oauth2Subject: String? = null,
        val firstName: String,
        val lastName: String,
        val email: String,
        val admin: Boolean,
        val signature: FileMeta?,
        val avatar: FileMeta?
) : BaseEntity()

File: backend-nest/src/users/User.entity.ts
import { Entity, ObjectID, ObjectIdColumn, Column } from 'typeorm';

@Entity()
export class HistorizedEntity<Entity> {
  @ObjectIdColumn()
  id: ObjectID;

  @Column()
  activeFrom: Date;

  @Column()
  entity: Entity;
}

@Entity()
export class User {
  @Column()
  id: string;

  @Column()
  sub?: string;

  @Column()
  firstName: string;

  @Column()
  lastName: string;

  @Column()
  email: string;

  @Column()
  admin: boolean;

  @Column()
  avatarUrl?: string;
}
